File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int square_of_sum(unsigned int number) {
    unsigned int ret = 0;

    for (unsigned int i = 1; i <= number; ++i) {
        ret += i;
    }

    return ret * ret;
}

unsigned int sum_of_squares(unsigned int number) {
    unsigned int ret = 0;

    for (unsigned int i = 1; i <= number; ++i) {
        ret += (i * i);
    }

    return ret;
}

unsigned int difference_of_squares(unsigned int number) {
    return square_of_sum(number) - sum_of_squares(number);
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	ret := 0

	for i := 1; i <= n; i++ {
		ret += i
	}

	return ret * ret
}

func SumOfSquares(n int) int {
	ret := 0

	for i := 1; i <= n; i++ {
		ret += i * i
	}

	return ret
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
